client/src/Tools/_framework/Paths/AllAssignmentStudentData.jsx

```diff
@@ -0,0 +1,113 @@
+import React, { useEffect, useState } from "react";
+import { useLoaderData } from "react-router";
+
+import {
+  TableContainer,
+  Table,
+  Thead,
+  Tr,
+  Th,
+  Tbody,
+  Td,
+  Link,
+  Tooltip
+} from "@chakra-ui/react";
+import axios from "axios";
+import { DoenetHeading as Heading } from "./Community";
+
+export async function action({ params, request }) {
+  return null;
+}
+
+export async function loader({ params }) {
+  const { data } = await axios.get(
+    `/api/getAllAssignmentStudentData/${params.userId}`,
+  );
+
+  let userId = Number(params.userId);
+
+  return {
+    userId,
+    studentScores: data.userScores,
+    assignments: data.assignments
+  };
+}
+
+export function AllAssignmentStudentData() {
+  const { userId, studentScores, assignments } =
+    useLoaderData();
+
+  useEffect(() => {
+    document.title = "Your Assignments";
+  });
+
+  const linkStyle = {
+    display: "block",
+    color: "var(--mainBlue)",
+  };
+
+  return (
+    <>
+        <Heading heading={"Your Assignments"} />
+
+        <Heading subheading="Score summary" />
+        <TableContainer>
+            <Table>
+            <Thead>
+                <Tr>
+                    <Th>Student</Th>
+                    {assignments.map((assignment) => {
+                        return (
+                            <Th key={`assignment${assignment.assignmentId}`}>
```

Comment:
Can you rotate this text so it is vertical? I have close to 100 assignments in some of my courses, so it'd be nice to have narrow columns.

Reply:
I've rotated it 315 degrees which I think is pretty readable. I'm a bit worried about some of the spacing values I hardcoded, but I've tried it out on different browsers and the mobile simulator and it seems to still look okay.